packages/git/src/browser/git-repository-provider.ts

```diff
@@ -45,38 +45,62 @@ export class GitRepositoryProvider {
         return this.onDidChangeRepositoryEmitter.event;
     }
 
-    protected fireOnDidChangeRepository(repository: Repository | undefined): void {
-        this.onDidChangeRepositoryEmitter.fire(repository);
-    }
-
     /**
      * Returns with all know repositories.
      */
     get allRepositories(): Repository[] {
         return this._allRepositories;
     }
 
-    /**
-     * Refreshes the state of this Git repository provider.
-     *  - Retrieves all known repositories from the backend, discards the current state of [all known repositories](GitRepositoryProvider.allRepositories).
-     *  - If no repository was [selected](GitRepositoryProvider.selectedRepository), then selects the first items from all known ones.
-     *  - If no repositories are available, leaves the selected one as `undefined`.
-     *  - If the previously selected one, does not exist among the most recently discovered one, selects the first one.
-     *  - This method blocks, if the workspace root is not yet set.
-     */
     async refresh(): Promise<void> {
```

Comment:
Wouldn't it a bit simpler to have a `limit` option. Which we set to `1` only when initially called from constructor and later always fetch all repos (replacing everything). I think we wouldn't need all the logic below nor the notion of a `containingRepository`.

Reply:
What should be returned if there is no containing repo, the first contained repo?

Reply:
And I think the logic will be needed anyway, otherwise it will flicker from one repository to several

Reply:
It should just return the first n results and then stop. Why would it flicker?

Reply:
It shows several repositories, you press refresh, it shows one repository, then it shows several repositories again.